lib/ansible/plugins/connection/__init__.py

```diff
@@ -37,6 +38,30 @@ def wrapped(self, *args, **kwargs):
     return wrapped
 
 
+def sanitise_become_output(become: BecomeBase, stdout: bytes, stderr: bytes) -> t.Tuple[bytes, bytes]:
+    """Helper function to remove the become success banner from stdout or stderr."""
+    if not become.success:
+        return stdout, stderr
+
+    raw = {'stdout': stdout, 'stderr': stderr}
+
+    for stream_type in ['stdout', 'stderr']:
+        success_idx = None
+        lines = raw[stream_type].split(b"\n")
```

Comment:
Doubling the memory usage seems potentially expensive when stdout/stderr is big. Have you considered using a generator-based streaming approach?

Reply:
that is one reason we should keep this in the connection code where we iterate over the network buffers

Reply:
The trouble is not all connection plugins actually iterate the network buffers. For example both paramiko and local might use a tty equivalent to do the initial become prompt once done it just retrieves the remaining data in one go using the transport equivalent. This might seem ideal but I then found out that the TTY/buffering work was only done if a become password was set. When no password was set the `BECOME-SUCCESS` message was being filtered out in the action code today.